src/main/java/com/example/server/Server.java

```diff
@@ -7,28 +7,37 @@
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.hibernate.SessionFactory;
+import com.example.server.localization.GetLocalizationsService;
+import com.example.server.localization.GetLocalizationsRepository;
 
-import java.net.http.HttpClient;
 import java.util.Objects;
 
 public class Server {
 
     private final LocalizationController localizationController;
+    //private final GetLocalizationsController getLocalizationsController;
 
     public Server() {
         ObjectMapper objectMapper = new ObjectMapper();
         objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         SessionFactory sessionFactory = HibernateUtils.getSessionFactory();
+
+        // Inicjalizacja dla dodawania lokalizacji i pobierania lokalizacji
         LocalizationRepository localizationRepository = new LocalizationRepository(sessionFactory);
         LocalizationService localizationService = new LocalizationService(localizationRepository);
-        LocalizationController localizationController = new LocalizationController(objectMapper, localizationService);
+        GetLocalizationsRepository getLocalizationsRepository = new GetLocalizationsRepository(sessionFactory);
+        GetLocalizationsService getLocalizationsService = new GetLocalizationsService(getLocalizationsRepository);
+        LocalizationController localizationController = new LocalizationController(objectMapper, localizationService, getLocalizationsService);
         this.localizationController = localizationController;
+
     }
 
     // mapujemy requesty HTTP na metody kontrolera
-    public String callServer(String method, String path, String json) {
+    public String callServer(String method, String path, String json) throws JsonProcessingException {
```

Comment:
JsonProcessingException is caught in the LocalizationController

Reply:
In IntelliJ, the method localizationController.getLocalizations() is not permitted to execute without handling the exception com.fasterxml.jackson.core.JsonProcessingException. It recommends including this exception in the method's signature to manage potential errors gracefully.